nucliadb/src/nucliadb/common/cluster/rebalance.py

```diff
@@ -435,6 +410,22 @@ def get_target_shard(
     return target_shard, capacity
 
 
+async def count_resources_in_shard(driver: Driver, kbid: str, shard_id: str) -> int:
+    driver = cast(PGDriver, driver)
+    async with driver._get_connection() as conn:
+        cur = conn.cursor("")
+        await cur.execute(
+            """
+            SELECT COUNT(*) FROM resources WHERE key ~ '/kbs/[^/]*/r/[^/]*/shard$' AND key ~ %s AND encode(value, 'escape') LIKE %s;
```

Comment:
```suggestion
            SELECT COUNT(*) FROM resources WHERE key ~ '/kbs/[^/]*/r/[^/]*/shard$' AND key ~ %s AND value = %s;
```

This is the same, but much faster